external/fv3fit/fv3fit/reservoir/model.py

```diff
@@ -49,38 +43,19 @@ def dump(self, path: str) -> None:
         Args:
             path: a URL pointing to a directory
         """
-
-        fs: fsspec.AbstractFileSystem = fsspec.get_fs_token_paths(path)[0]
-        fs.makedirs(path, exist_ok=True)
-        mapper = fs.get_mapper(path)
-
-        mapper[self._READOUT_NAME] = self.readout.dumps()
-        metadata = {
-            "reservoir_hyperparameters": dataclasses.asdict(
-                self.reservoir.hyperparameters
-            )
-        }
-        mapper[self._METADATA_NAME] = yaml.safe_dump(metadata).encode("UTF-8")
+        self.readout.dump(path)
+        self.reservoir.dump(path)
```

Comment:
Is it safe to dump these both to the same path? This class doesn't know the low-level details of how these subcomponents are stored, it's very possible for them to re-use the same filenames (especially if someone modifies them after you).

Reply:
See above comment- changed to dump to readout/reservoir subdirectories.